host-controller/src/main/java/org/jboss/as/host/controller/ManagedServer.java

```diff
@@ -814,39 +844,68 @@ public boolean execute(ManagedServer server) throws Exception {
 
     private class ServerStopTask implements TransitionTask {
 
-        private final int permit;
-        private final int timeout;
+        private final int gracefulTimeout;
+        private final BlockingTimeout blockingTimeout;
 
-        private ServerStopTask(int permit, int timeout) {
-            this.permit = permit;
-            this.timeout = timeout;
+        private ServerStopTask(int gracefulTimeout) {
+            this.gracefulTimeout = gracefulTimeout;
+            this.blockingTimeout = null;
+        }
```

Comment:
An alternative here instead of null is make BlockingTimeoutImpl public and just use BlockingTimeoutImpl(null)